### PR TITLE
Support channel access token v2.1 API

### LINKED ISSUE #254: Support Channel access token v2.1
https://developers.line.biz/en/news/2020/04/06/channel-access-token-apis-v2-1/

### PR DESCRIPTION
**Pull request by morugu**
release: https://developers.line.biz/en/news/2020/04/06/channel-access-token-apis-v2-1/
docs: https://developers.line.biz/en/reference/messaging-api/#oauth

- implement new functions `issue_channel_access_token_v2_1`, `revoke_channel_access_token_v2_1`, `get_channel_access_tokens_v2_1` in `linebot/api.py`

- implement a new response `ChannelAccessTokens` in `linebot/models/responses.py`

Closes #254

### CODE REVIEW

**Comment by okue** [comment]
Could you check the CI result?

File: linebot/api.py [hunk 1]
```diff
@@ -1034,6 +1034,88 @@ def get_insight_message_event(self, request_id, timeout=None):
...
 
+    def issue_channel_access_token_v2_1(
+            self, client_assertion, grant_type='client_credentials',
+            client_assertion_type='urn:ietf:params:oauth:client-assertion-type:jwt-bearer',
+            timeout=None):
+        """Issues a channel access token v2.1.
+
+        https://developers.line.biz/en/reference/messaging-api/#issue-channel-access-token-v2-1
+
+        :param str client_assertion: Clint assertion.
```
Comment (cryeo) [inline linebot/api.py:1045]: [nits]
`Clint` -> `Client`

File: linebot/api.py [hunk 2]
```diff
@@ -1034,6 +1034,88 @@ def get_insight_message_event(self, request_id, timeout=None):
...
+
+    def get_channel_access_tokens_v2_1(
+            self, client_assertion,
+            client_assertion_type='urn:ietf:params:oauth:client-assertion-type:jwt-bearer',
+            timeout=None):
+        """Get issued channel access tokens v2.1.
+
+        https://developers.line.biz/en/reference/messaging-api/#get-issued-channel-access-tokens-v2-1
+
+        :param str client_assertion: Clint assertion.
```
Comment (cryeo) [inline linebot/api.py:1101]: [nits]
`Clint` -> `Client`

File: linebot/api.py [hunk 3]
```diff
@@ -1034,6 +1034,88 @@ def get_insight_message_event(self, request_id, timeout=None):
...
+        :param str grant_type: `client_credentials`
+        :param str client_assertion_type: `urn:ietf:params:oauth:client-assertion-type:jwt-bearer`.
+        :param timeout: (optional) How long to wait for the server
+            to send data before giving up, as a float,
+            or a (connect timeout, read timeout) float tuple.
+            Default is self.http_client.timeout
+        :type timeout: float | tuple(float, float)
+        :rtype: :py:class:`linebot.models.responses.IssueChannelTokenResponse`
+        :return: IssueChannelTokenResponse instance
```
Comment (cryeo) [inline linebot/api.py:1054]: [imo]
I think that the return description is unnecessary for consistency with other descriptions.

File: linebot/api.py [hunk 4]
```diff
@@ -1034,6 +1034,88 @@ def get_insight_message_event(self, request_id, timeout=None):
...
+
+        :param str client_assertion: Clint assertion.
+        :param str client_assertion_type: `urn:ietf:params:oauth:client-assertion-type:jwt-bearer`.
+        :param timeout: (optional) How long to wait for the server
+            to send data before giving up, as a float,
+            or a (connect timeout, read timeout) float tuple.
+            Default is self.http_client.timeout
+        :type timeout: float | tuple(float, float)
+        :rtype: :py:class:`linebot.models.responses.ChannelAccessTokens`
+        :return: ChannelAccessTokens instance
```
Comment (cryeo) [inline linebot/api.py:1109]: [imo]
I think that the return description is unnecessary for consistency with other descriptions. (`rtype` alone is enough)

File: linebot/api.py (hunk 2 above)
Comment (cryeo) [inline linebot/api.py:1101]: [nits]
`Clint` -> `Client`

File: linebot/api.py (hunk 3 above)
Comment (cryeo) [inline linebot/api.py:1054]: [imo]
I think that the return description is unnecessary for consistency with other descriptions. (`rtype` alone is enough)

File: linebot/api.py (hunk 1 above)
Comment (cryeo) [inline linebot/api.py:1045]: [nits]
`Clint` -> `Client`

File: linebot/api.py [hunk 5]
```diff
@@ -1034,6 +1034,88 @@ def get_insight_message_event(self, request_id, timeout=None):
...
+            data={
+                'grant_type': grant_type,
+                'client_assertion_type': client_assertion_type,
+                'client_assertion': client_assertion,
+            },
+            headers={'Content-Type': 'application/x-www-form-urlencoded'},
+            timeout=timeout
+        )
+
+        return IssueChannelTokenResponse.new_from_json_dict(response.json)
```
Comment (cryeo) [inline linebot/api.py:1067]: [FYI]
Due to [this update](https://developers.line.biz/en/news/2020/06/22/new-channel-access-token-endpoint-and-deprecation/), the response of issue channel token via v2.1 contains `key_id` property.

So, `IssueChannelTokenResponse` should be updated to deal with `key_id`.

**Comment by louis70109** [comment]
@morugu could you fix this PR to let SDK advance to 1.17?
Thanks in advance🙂.

reference milestone: https://github.com/line/line-bot-sdk-python/milestone/1

**Pull request closed without merging**